contracts/ronin/staking/DelegatorStaking.sol

```diff
@@ -150,7 +150,15 @@ abstract contract DelegatorStaking is BaseStaking, IDelegatorStaking {
   ) private notPoolAdmin(_pool, _delegator) {
     if (_amount == 0) revert ErrUndelegateZeroAmount();
     if (_pool.delegatingAmount[_delegator] < _amount) revert ErrInsufficientDelegatingAmount();
-    if (_pool.lastDelegatingTimestamp[_delegator] + _cooldownSecsToUndelegate >= block.timestamp) {
+
+    bool _candidateRenounced;
+    if (_validatorContract.isValidatorCandidate(_pool.addr)) {
+      _candidateRenounced = _validatorContract.getCandidateInfo(_pool.addr).revokingTimestamp != 0;
+    }
```

Comment:
Does this mean when a candidate is removed, the delegators still have to wait until cooldown?

Reply:
The suggested change may cause `revert` since `getCandidateInfo()` has a assertion of `_pool.addr` must be a candidate. Not tested on short-circuit, but should not take risk on this.

Reply:
Fixed both concerns. Check out the short-circuiting rule [here](https://docs.soliditylang.org/en/latest/types.html#booleans).